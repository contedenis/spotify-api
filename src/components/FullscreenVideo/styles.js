// @packages
import styled from 'styled-components';

const FullscreenVideoStyled = styled.video`
  height: auto;
  left: 50%;
  min-height: 100%;
  min-width: 100%;
  position: fixed;
  top: 50%;
  transform: translate(-50%, -50%);
  width: auto;
  z-index: -100;
`;

export { FullscreenVideoStyled };
