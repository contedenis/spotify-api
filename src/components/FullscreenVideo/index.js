// @packages
import React from 'react';
import PropTypes from 'prop-types';

// @own
import { FullscreenVideoStyled } from './styles';

function FullscreenVideo({
  poster,
  src,
  ...props
}) {
  return (
    <FullscreenVideoStyled
      poster={poster}
      autoPlay
      muted
      loop
      {...props}
    >
      <source src={src} type="video/webm" />
      <source src={src} type="video/mp4" />
    </FullscreenVideoStyled>
  );
}

FullscreenVideo.defaultProps = {
  poster: '',
};

FullscreenVideo.propTypes = {
  poster: PropTypes.string,
  src: PropTypes.string.isRequired,
};

export default FullscreenVideo;
