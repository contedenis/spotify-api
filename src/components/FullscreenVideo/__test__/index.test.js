// @packages
import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

// @own
import FullscreenVideo from '../index';

describe('FullscreenVideo', () => {
  it('matches snapshot', () => {
    const { container } = render(
      <FullscreenVideo />,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
