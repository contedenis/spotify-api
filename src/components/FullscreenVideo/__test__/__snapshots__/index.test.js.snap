// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FullscreenVideo matches snapshot 1`] = `
<video
  autoplay=""
  class="sc-AxjAm dBmncd"
  loop=""
  poster=""
>
  <source
    type="video/webm"
  />
  <source
    type="video/mp4"
  />
</video>
`;
